### ISSUE TITLE
Mypy finds a syntax error on unpack operators with Python 3.11

### ISSUE THREAD

**Issue by sdiebolt** [python/mypy#16663]
**Bug Report**

The following code runs correctly with Python 3.11:

```{python}
import numpy as np

test = np.zeros((5, 5, 5))
test[*(1, 2, 3)]
```

But mypy finds a syntax error on line 4 because of the unpack operator. Note that no error is found with Python 3.12.

**Expected Behavior**

No error is found.

**Actual Behavior**

```
error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

**Your Environment**

- Mypy version used: 1.7.1
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.11

**Comment by hauntsaninja** [comment]
It checks just fine for me. Note you need to have installed mypy with Python 3.11 or newer in order to be able to type check Python 3.11 code

**Comment by sdiebolt** [comment]
@hauntsaninja Actually I had this problem from a fresh 3.11 virtual env. But I just noticed that I had another version of mypy in my PATH that bypassed the one installed in the virtual env! Sorry to have bothered you. :disappointed:
